client/ayon_premiere/hooks/pre_launch_install_ayon_extension.py

```diff
@@ -0,0 +1,105 @@
+    """
+
+    app_groups = {"premiere"}
+
+    order = 1
+    launch_types = {LaunchTypes.local}
+
+    def execute(self):
+        try:
+            settings = self.data["project_settings"][self.host_name]
+            if not settings["hooks"]["InstallAyonExtensionToPremiere"][
+                "enabled"
+            ]:
+                return
+            self.inner_execute()
+
+        except Exception:
+            self.log.warning(
+                "Processing of {} crashed.".format(self.__class__.__name__),
+                exc_info=True,
+            )
+
+    def inner_execute(self):
+        self.log.info("Installing AYON Premiere extension.")
+
+        # Windows only for now.
+        if not platform.system().lower() == "windows":
+            self.log.info("Non Windows platform. Cancelling..")
+            return
+
+        target_path = os.path.join(
+            os.environ["appdata"], r"Adobe\CEP\extensions\io.ynput.PPRO.panel"
+        )
+
+        extension_path = os.path.join(
+            PREMIERE_ADDON_ROOT,
+            r"api\extension.zxp",
+        )
+
+        try:
+            deploy = self._deploy_extension(extension_path, target_path)
+
+            if not deploy:
+                self.log.debug(f"Extension already deployed at {target_path}.")
+                return
+
+            self.log.debug(f"Creating directory: {target_path}")
+            os.makedirs(target_path, exist_ok=True)
+
+            with ZipFile(extension_path, "r") as zip:
+                zip.extractall(path=target_path)
+
+            self.log.info("Successfully installed AYON extension")
+
+        except OSError as error:
+            self.log.warning(f"OS error has occured: {error}")
+
+        except PermissionError as error:
+            self.log.warning(f"Permissions error has occured: {error}")
+
+        except Exception as error:
+            self.log.warning(f"An unexpected error occured: {error}")
+
+
+    def _deploy_extension(self, extension_path, target_path):
+        """Check if current extension version is installed, purge old ones."""
+        with ZipFile(extension_path, 'r') as zip_file:
+            with zip_file.open('CSXS/manifest.xml') as manifest_file:
+                content = manifest_file.read()
+                pattern = r'ExtensionBundleVersion="([^"]+)"'
+                extension_version = re.search(pattern, str(content)).group(1)
+
+        deployed_manifest_path = os.path.join(
+            target_path, "CSXS", "manifest.xml")
+        if not os.path.exists(deployed_manifest_path):
+            return True
+
+        with open(deployed_manifest_path, "r") as deployed_manifest_file:
+            deployed_content = deployed_manifest_file.read()
+            deployed_extension_version =(
+                re.search(pattern, str(deployed_content)).group(1)
+            )
+
+        if extension_version != deployed_extension_version:
+            self.log.info(f"Purging {target_path} because different version")
```

Comment:
I wonder if it'd be helpful for the logs to include info of what the purged version's version number was and also what the new version number being deployed will be?